akka-typed/src/main/scala/akka/typed/ActorContext.scala

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (C) 2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.typed
+
+import scala.concurrent.duration.Duration
+import scala.collection.immutable
+import scala.collection.immutable.TreeSet
+import scala.collection.immutable.TreeMap
+import akka.util.Helpers
+import akka.{ actor ⇒ untyped }
+import scala.reflect.ClassTag
+import scala.reflect.classTag
+import java.util.concurrent.ConcurrentLinkedQueue
+import scala.annotation.tailrec
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.ExecutionContextExecutor
+import akka.typed.Behavior.stoppedBehavior
+
+/**
+ * An Actor is given by the combination of a [[Behavior]] and a context in
+ * which this behavior is executed. As per the Actor Model an Actor can perform
+ * the following actions when processing a message:
+ *
+ *  - send a finite number of messages to other Actors it knows
+ *  - create a finite number of Actors
+ *  - designate the behavior for the next message
+ *
+ * In Akka the first capability is accessed by using the `!` or `tell` method
+ * on an [[ActorRef]], the second is provided by [[ActorContext#spawn]]
+ * and the third is implicit in the signature of [[Behavior]] in that the next
+ * behavior is always returned from the message processing logic.
+ *
+ * An `ActorContext` in addition provides access to the Actor’s own identity (“`self`”),
+ * the [[ActorSystem]] it is part of, methods for querying the list of child Actors it
+ * created, access to [[Terminated DeathWatch]] and timed message scheduling.
+ */
+trait ActorContext[T] {
+
+  /**
+   * The identity of this Actor, bound to the lifecycle of this Actor instance.
+   * An Actor with the same name that lives before or after this instance will
+   * have a different [[ActorRef]].
+   */
+  def self: ActorRef[T]
+
+  /**
+   * The [[Props]] from which this Actor was created.
+   */
+  def props: Props[T]
+
+  /**
+   * The [[ActorSystem]] to which this Actor belongs.
+   */
+  def system: ActorSystem[Nothing]
+
+  /**
+   * The list of child Actors created by this Actor during its lifetime that
+   * are still alive, in no particular order.
+   */
+  def children: Iterable[ActorRef[Nothing]]
+
+  /**
+   * The named child Actor if it is alive.
+   */
+  def child(name: String): Option[ActorRef[Nothing]]
+
+  /**
+   * Create a child Actor from the given [[Props]] under a randomly chosen name.
+   * It is good practice to name Actors wherever practical.
+   */
+  def spawnAnonymous[U](props: Props[U]): ActorRef[U]
+
+  /**
+   * Create a child Actor from the given [[Props]] and with the given name.
+   */
+  def spawn[U](props: Props[U], name: String): ActorRef[U]
```

Comment:
Did we switch to 'spawn' just to be backward compatible and not clash? Else, why the change? 'actorOf' is pretty well established. 'spawn' sounds like Erlang-envy. Do we need to support untyped Actors here? For interop?